lib/types.nix

```diff
@@ -333,7 +333,15 @@ rec {
       name = "either";
       description = "${t1.description} or ${t2.description}";
       check = x: t1.check x || t2.check x;
-      merge = mergeOneOption;
+      merge = loc: defs:
+        let
+          defList = fold (d: acc: acc ++ [ d.value ]) [] defs;
```

Comment:
This is simpler as `map (d: d.value)`, provided the ordering is irrelevant (which it looks to be below).